changes/2024-TBD_key-store-admin/background.md

```diff
@@ -0,0 +1,224 @@
+maintained and documented by Crypto Tools.
+
+However,
+this approach forces the Material Provider's Client
+to directly reference DynamoDB.  
+In the long run,
+Crypto Tools MAY want allow MPL Consumers to optionally
+depend on the AWS SDK,
+as the original AWS Encryption SDK afforded that.
+Adding a direct reference to DynamoDB
+to the Material Provider's Client
+complicates such an ambition.
+
+# Name of the Client?
+
+It is best that we bike shed on this early.
+
+The proposed name for this client is
+`the Key Store Admin Client`
+though we may often refer to it as
+just the `Key Store Admin`.
+
+# Proposed Construction
+
+The Key Store Admin requires the following arguments:
+
+- Storage Reference
+- Logical Key Store Name
+
+See [Key Store Persistence Changes](changes/2024-6-17_key-store-persistance/background.md) for details
+on Storage Reference.
+
+# Common Parameters and their defaults
+
+## `KMS Identifier`
+
+`KMS Identifier` is a KMS ARN which identifies the KMS Key
+that will be used for the operation.
+This ARN MUST NOT be an Alias.
+This ARN MUST be a valid
+[AWS KMS Key ARN](./aws-kms/aws-kms-key-arn.md#a-valid-aws-kms-arn).
+This ARN MAY be a Multi-Region Key (MRK) or Single Region Key.
+
+## Key Management
+
+`Key Management` is a union,
+such that additional options maybe added at a later date.
+
+Members of `Kms Management` are elements that
+authorize the use of Branch Key Cryptographic Materials
+by decrypting them in a manner that
+authenticates all attributes of the Branch Key.
+
+They are also used to create Branch Key Encrypted Cryptographic Materials,
+and when doing so MUST cryptographically bind all attributes of the Branch Key
+to the cipher-text.
+
+Currently, there is only one `Key Management` option.
+
+`Key Management` is never used directly by the Key Store Admin.
+
+However, it was introduced into the Key Store
+as part of the [Key Store Persistence Changes](changes/2024-6-17_key-store-persistance/background.md).
+It is, at least at this time,
+not visible to any MPL Consumer facing operations.
+
+### `AwsKms`
+
+`AwsKms` is the only `Key Management` option currently
+supported by the Key Store or Key Store Admin.
+
+This option is provided by default,
+if no other `Key Management` is provided.
+
+`AwsKms` represents AWS Key Management Service (KMS).
+
+The structure has two optional fields:
+
+- `grantTokenList`, a list of strings
+- `kmsClient`, a reference to an AWS SDK KMS client
+
+If no `grantTokenList` is provided,
+it defaults to an empty list.
+
+If no `kmsClient` is provided,
+the default KMS client is constructed via the AWS SDK.
+
+## Key Management Strategy
+
+`Key Management Strategy` is a union,
+such that additional options maybe added at a later date.
+
+`Key Management Strategy` determines which Operations
+of a `Key Management` are used by the Client.
+
+For example,
+an MPL Consumer MAY want to avoid calling `kms:ReEncrypt`,
+and would rather use `kms:Decrypt` followed by `kms:Encrypt`
```

Comment:
Note: We should also consider using GDK in this case.
Because if you are using this option because you do not want to use ask for new access, then you would likely have Encrypt/Decrypt/GenerateDataKey